demisto_sdk/commands/setup_env/setup_environment.py

```diff
@@ -92,27 +94,180 @@ def get_docker_python_path(docker_prefix: str) -> List[str]:
+            )
+            logger.info(
+                f"Original configuration file was backed up to '{backup_file_path}'."
+            )
+
+        config_data.write(
+            str(file_path), pretty_print=True, xml_declaration=True, encoding="utf-8"
+        )
+        logger.info(
+            f"Configuration file ('{file_path}') was successfully configured for automatic module discovery. "
+            f"New entries added: {added_entries_count}."
+        )
+
+    else:
+        logger.info(
+            f"All entries are already configured on the configuration file ('{file_path}'). No changes were made."
+        )
 
 
-def configure_dotenv():
+def update_pycharm_config_xml_data(
+    config_data: etree._ElementTree, python_discovery_paths: List[Path]
+) -> int:
     """
-    This functions configures the .env file located with PYTHONPATH and MYPYPATH
-    This is needed for discovery and linting for CommonServerPython, demistomock and API Modules files.
+    Configure and update the XML data within the configuration file
+    to add the given python paths to the module discovery.
+
+    Args:
+        config_data (etree._ElementTree): The XML data to update
+        python_discovery_paths (List[Path]): The python paths to add to the module discovery
+
+    Returns:
+        int: The number of added entries
     """
-    DOTENV_PATH.touch()
-    python_path_values = ":".join((str(path) for path in PYTHONPATH))
-    mypy_path_values = ":".join(
-        (str(path) for path in PYTHONPATH if "site-packages" not in str(path))
+    url_prefix = "file://$MODULE_DIR$"
+    module_root_manager_name = "NewModuleRootManager"
+    module_root_manager_component = f"component[@name='{module_root_manager_name}']"
+    module_root_manager_content = (
+        f"{module_root_manager_component}/content[@url='{url_prefix}']"
     )
-    update_dotenv({"PYTHONPATH": python_path_values, "MYPYPATH": mypy_path_values})
+
+    # Generate the module root manager component if it doesn't exist
+    if config_data.find(module_root_manager_component) is None:
+        # Add spaces following last component
+        root_data = config_data.getroot()
+
+        module_root_manager_component_data = etree.SubElement(root_data, "component")
+        module_root_manager_component_data.set("name", module_root_manager_name)
+
+    # Generate the content component if it doesn't exist
+    if config_data.find(module_root_manager_content) is None:
+        # Add spaces following last content item
+        module_root_manager_component_data = config_data.find(
+            module_root_manager_component
+        )
+
+        etree.SubElement(module_root_manager_component_data, "content", url=url_prefix)
+
+    source_folders = config_data.findall(module_root_manager_content + "/sourceFolder")
+    existing_paths = set()
+
+    for source_folder in source_folders:
+        if url := source_folder.get("url"):
```

Comment:
what is the url?

Reply:
For example, for:
```xml
<component name="NewModuleRootManager">
  <content url="file://$MODULE_DIR$">
    <sourceFolder url="file://$MODULE_DIR$/test1/test2" isTestSource="false"/>
    <sourceFolder url="file://$MODULE_DIR$/test3/test4" isTestSource="false"/>
  </content>
</component>
```
A `source_folder` would be:
```xml
<sourceFolder url="file://$MODULE_DIR$/test1/test2" isTestSource="false"/>
```
(And `url` would be the `url` value, representing the path to the module to add as a source)